apps/jobs/views.py

```diff
@@ -745,3 +745,40 @@ def update_submission(request, challenge_pk):
     submission.save()
     response_data = {'success': 'Submission result has been successfully updated'}
     return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(['GET', ])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_challenge_submissions_by_status(request, challenge_pk):
+
+    challenge = get_challenge_model(challenge_pk)
+
+    if not is_user_a_host_of_challenge(request.user, challenge.id):
+        response_data = {'error': 'Sorry, you are not authorized to make this request!'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    submission_status = request.query_params.get('status', None)
+
+    valid_submission_status = [Submission.SUBMITTED, Submission.RUNNING, Submission.FAILED,
+                               Submission.CANCELLED, Submission.FINISHED, Submission.SUBMITTING]
+
+    if submission_status not in valid_submission_status:
+        response_data = {
+            'error': 'Invalid submission status {}'.format(submission_status)
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    submissions_done_in_challenge = Submission.objects.filter(
+        challenge_phase__challenge=challenge.id, status=submission_status)
+
+    submissions_done_in_challenge_count = submissions_done_in_challenge.count()
+
+    serializer = SubmissionSerializer(
+        submissions_done_in_challenge, many=True, context={'request': request})
+    response_data = {
+        'submissions': serializer.data,
+        'submissions_count': submissions_done_in_challenge_count
```

Comment:
Please remove the submissions_count since you can always get it from `len(serializer.data)`